packages/clients/src/aptos/wallet.ts

```diff
@@ -0,0 +1,431 @@
+/**
+ * Copyright 2023-2024 Shinami Corp.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SimpleTransaction,
+  MultiAgentTransaction,
+  AnyRawTransaction,
+  AccountAuthenticator,
+  PendingTransactionResponse,
+} from "@aptos-labs/ts-sdk";
+import { Infer, array, integer, object, string, unknown } from "superstruct";
+import { ShinamiRpcClient, errorDetails, trimTrailingParams } from "../rpc.js";
+import { JSONRPCError } from "@open-rpc/client-js";
+
+const KEY_RPC_URL = "https://api.shinami.com/aptos/key/v1";
+const WALLET_RPC_URL = "https://api.shinami.com/aptos/wallet/v1";
+
+/**
+ * Shinami Key RPC client for Aptos.
+ */
+export class KeyClient extends ShinamiRpcClient {
+  /**
+   * @param accessKey Aptos Wallet access key.
+   * @param url Optional URL override.
+   */
+  constructor(accessKey: string, url: string = KEY_RPC_URL) {
+    super(accessKey, url);
+  }
+
+  /**
+   * Creates a session with the provided wallet secret.
+   * @param secret Wallet secret. For later operations to succeed, the secret must match what the
+   *    wallet was created with.
+   * @returns A session token used to perform sensitive wallet operations with, valid for 10 mins.
+   */
+  createSession(secret: string): Promise<string> {
+    return this.request("key_createSession", [secret], string());
+  }
+}
+
+/**
+ * Data wrapper for a Shinami Aptos wallet.
+ */
+const WalletInfo = object({
+  accountAddress: string(),
+});
+type WalletInfo = Infer<typeof WalletInfo>;
+
+/**
+ * Transaction signing result
+ */
+const SignTransactionResult = object({
+  signature: array(integer()),
+});
+type SignTransactionResult = Infer<typeof SignTransactionResult>;
+
+/**
+ * Gasless transaction execution result
+ */
+const ExecuteGaslessTransactionResult = object({
+  pendingTransaction: unknown(),
+});
+type ExecuteGaslessTransactionResult = Infer<
+  typeof ExecuteGaslessTransactionResult
+>;
+
+/**
+ * Shinami Wallet RPC client for Aptos.
+ */
+export class WalletClient extends ShinamiRpcClient {
+  /**
+   * @param accessKey Aptos Wallet access key.
+   * @param url Optional URL override.
+   */
+  constructor(accessKey: string, url: string = WALLET_RPC_URL) {
+    super(accessKey, url);
+  }
+
+  /**
+   * Creates a new wallet that is not initialized on chain.
+   * @param walletId Wallet id. Must not have been previously used, or an error will be returned.
+   * @param sessionToken Session token. Obtained by `KeyClient.createSession`.
+   * @returns Aptos address of the created wallet.
+   */
+  async createWallet(
+    walletId: string,
+    sessionToken: string,
+  ): Promise<WalletInfo> {
+    return this.request(
```

Comment:
For this and the other methods that return an AccountAddress, it would be nice if we unpack it and return a string, like we do with ShinamiWalletSigner (and SUI, though the difference there starts with the Sui Wallet Service returning a string and not an object). There may be something I'm overlooking, but for user convenience and for consistency, it would be nice to just return the string.

Reply:
Yea this is the same as @jasonxh 's comments. I'll be making the change to unpack it directly to the aptos sdk AccountAddress type (rather than just a string)

Reply:
I think @aaronTufts was asking for the exact opposite, which I don't agree with. The new design we are going with is to integrate as seamlessly with the rest of the SDK as possible, so that using a `ShinamiWalletSigner` feels like using a builtin signer without unnecessary pre or post processing. The difference can be easily seen from the updated examples. If the user really needs a string, Aptos SDK already makes it very easy, and I assume the user is already familiar with the steps because they use Aptos SDK way more than Shinami SDK.

Reply:
For clarity: I prefer where we ended up.

When I said "AccountAddress" I misspoke and did not mean to refer to the Aptos type, but instead to the string we were returning, which I believe was called "accountAddress". I meant to ask for consistency, not a string vs something else. Sorry for the confusion from lack of precision.